<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>syntR Tutorial • syntR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="syntR Tutorial">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">syntR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/syntr_tutorial.html">syntR Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>syntR Tutorial</h1>
                        <h4 class="author">Kate Ostevik and Kieran Samuk</h4>
            
            <h4 class="date">2019-02-18</h4>
      
      
      <div class="hidden name"><code>syntr_tutorial.Rmd</code></div>

    </div>

    
    
<div id="what-is-syntr" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-syntr" class="anchor"></a>What is syntR?</h2>
<p>syntR is an R package for identifying synteny blocks shared between two genetic maps. The core algorithm implemented in the package focuses on identifying synteny blocks via comparison of marker orders (e.g. from genetic maps of different species or populations). syntR identifies synteny blocks using a clustering algorithm tuned to the linear nature of genetic map data and generates an annotated list synteny block membership for each marker. syntR does not identify segmental duplications.</p>
</div>
<div id="requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h2>
<p>syntR requires marker positions (chromosome and map position for each markers) from two genetic maps. The marker orders in the two maps need to be separately inferred for each map prior to using syntR. If you need to build you maps prior to using syntR, we reccomend looking into <a href="http://www.rqtl.org/">R/qtl</a>, <a href="https://cran.r-project.org/web/packages/ASMap/index.html">ASMap</a>, and <a href="https://sourceforge.net/p/lep-map3/wiki/LM3%20Home/">Lep-MAP</a>.</p>
</div>
<div id="input-file" class="section level2">
<h2 class="hasAnchor">
<a href="#input-file" class="anchor"></a>Input File</h2>
<p>The input file has six columns. Each row represents the map position of a single marker (e.g. a SNP, a microsatillite, etc.) in two different maps.</p>
<table class="table">
<caption>The example input file</caption>
<thead><tr class="header">
<th align="left">map1_name</th>
<th align="left">map1_chr</th>
<th align="right">map1_pos</th>
<th align="left">map2_name</th>
<th align="left">map2_chr</th>
<th align="right">map2_pos</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">HanXRQChr12_10463750</td>
<td align="left">Ann12</td>
<td align="right">23.33200</td>
<td align="left">HanXRQChr12_10463750</td>
<td align="left">Pet12_16</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">HanXRQChr12_10463069</td>
<td align="left">Ann12</td>
<td align="right">23.33091</td>
<td align="left">HanXRQChr12_10463069</td>
<td align="left">Pet12_16</td>
<td align="right">3.660274</td>
</tr>
<tr class="odd">
<td align="left">HanXRQChr12_67274477</td>
<td align="left">Ann12</td>
<td align="right">51.14234</td>
<td align="left">HanXRQChr12_67274477</td>
<td align="left">Pet12_16</td>
<td align="right">3.660274</td>
</tr>
<tr class="even">
<td align="left">HanXRQChr12_11545866</td>
<td align="left">Ann12</td>
<td align="right">25.01880</td>
<td align="left">HanXRQChr12_11545866</td>
<td align="left">Pet12_16</td>
<td align="right">3.660274</td>
</tr>
<tr class="odd">
<td align="left">HanXRQChr12_11545707</td>
<td align="left">Ann12</td>
<td align="right">25.01856</td>
<td align="left">HanXRQChr12_11545707</td>
<td align="left">Pet12_16</td>
<td align="right">3.660274</td>
</tr>
<tr class="even">
<td align="left">HanXRQChr12_9937279</td>
<td align="left">Ann12</td>
<td align="right">22.47337</td>
<td align="left">HanXRQChr12_9937279</td>
<td align="left">Pet12_16</td>
<td align="right">4.567248</td>
</tr>
</tbody>
</table>
<p>The columns contents are as follows:</p>
<ul>
<li>
<strong>map1_name</strong>: A unique identifier for the marker. In this case, the markers are named based on where they reside in the sunflower reference genome (determined by aligning GBS reads).</li>
<li>
<strong>map1_chr</strong>: The chromosome on which the marker resides in the first map</li>
<li>
<strong>map1_pos</strong>: The map position (e.g. in cM) of the marker in the first map</li>
<li>
<strong>map2_name</strong>: As above, but for map2. These could be identical to the map1_names, or not, depending on your marker classification scheme.</li>
<li>
<strong>map2_chr</strong>: The chromosome on which the marker resides in the second map</li>
<li>
<strong>map2_pos</strong>: The map position of the marker in the second map</li>
</ul>
</div>
<div id="example-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#example-workflow" class="anchor"></a>Example Workflow</h2>
<p>Below, we work through an example using genetic map data from two sunflower species. Note that some of the details and tuning parameters will require tailoring to your specific maps.</p>
</div>
<div id="load-and-format-the-map-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-and-format-the-map-data" class="anchor"></a>1. Load and format the map data</h2>
<p>We begin by loading the example data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install the package if needed!</span>
<span class="co"># devtools::install_github("ksamuk/syntR")</span>

<span class="co"># load the syntR package</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"syntR"</span>)

<span class="co"># load the example marker data</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(ann_pet_map)
<span class="kw">as_tibble</span>(ann_pet_map)
<span class="co">#&gt; # A tibble: 1,063 x 6</span>
<span class="co">#&gt;    map1_name          map1_chr map1_pos map2_name         map2_chr map2_pos</span>
<span class="co">#&gt;    &lt;fct&gt;              &lt;fct&gt;       &lt;dbl&gt; &lt;fct&gt;             &lt;fct&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 HanXRQChr12_10463… Ann12       23.3  HanXRQChr12_1046… Pet12_16     0   </span>
<span class="co">#&gt;  2 HanXRQChr12_10463… Ann12       23.3  HanXRQChr12_1046… Pet12_16     3.66</span>
<span class="co">#&gt;  3 HanXRQChr12_67274… Ann12       51.1  HanXRQChr12_6727… Pet12_16     3.66</span>
<span class="co">#&gt;  4 HanXRQChr12_11545… Ann12       25.0  HanXRQChr12_1154… Pet12_16     3.66</span>
<span class="co">#&gt;  5 HanXRQChr12_11545… Ann12       25.0  HanXRQChr12_1154… Pet12_16     3.66</span>
<span class="co">#&gt;  6 HanXRQChr12_99372… Ann12       22.5  HanXRQChr12_9937… Pet12_16     4.57</span>
<span class="co">#&gt;  7 HanXRQChr12_33608… Ann12        9.06 HanXRQChr12_3360… Pet12_16     4.57</span>
<span class="co">#&gt;  8 HanXRQChr12_13257… Ann12       27.4  HanXRQChr12_1325… Pet12_16     7.22</span>
<span class="co">#&gt;  9 HanXRQChr12_17054… Ann12       32.0  HanXRQChr12_1705… Pet12_16    12.6 </span>
<span class="co">#&gt; 10 HanXRQChr12_17071… Ann12       32.0  HanXRQChr12_1707… Pet12_16    12.6 </span>
<span class="co">#&gt; # … with 1,053 more rows</span>

<span class="co"># read in list of chromosome lengths for map1</span>
<span class="co"># this is an optional file that defines the maximum </span>
<span class="co"># length of each chromosome (in cM) if this is known</span>
<span class="co"># NB: this is useful when the markers being compared </span>
<span class="co"># do not cover the entire map</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(ann_chr_lengths)
ann_chr_lengths
<span class="co">#&gt;   map1_chr map1_max_chr_lengths</span>
<span class="co">#&gt; 1    Ann08                63.68</span>
<span class="co">#&gt; 2    Ann09                96.46</span>
<span class="co">#&gt; 3    Ann12                71.68</span>
<span class="co">#&gt; 4    Ann15                80.29</span>
<span class="co">#&gt; 5    Ann16               103.49</span>
<span class="co">#&gt; 6    Ann17                97.14</span>

<span class="co"># make_one_map places all the markers on a single scale</span>
<span class="co"># and adds padding between the chromosomes </span>
<span class="co"># to aid the algorithm and for visualization purposes</span>
map_list &lt;-<span class="st"> </span><span class="kw"><a href="../reference/make_one_map.html">make_one_map</a></span>(ann_pet_map, <span class="dt">map1_max_chr_lengths =</span> ann_chr_lengths)</code></pre></div>
<p><code>make_one_map</code> outputs a list object containing three elements: - [[1]]: the marker data file with 2 additional ordering columns - [[2]] and [[3]]: the locations of the chromosome breaks between map 1 and map 2</p>
</div>
<div id="reordering-the-map" class="section level2">
<h2 class="hasAnchor">
<a href="#reordering-the-map" class="anchor"></a>2. Reordering the map</h2>
<p>We can use <code><a href="../reference/plot_maps.html">plot_maps()</a></code> to create a basic dot plot of our marker data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># plot map</span>
<span class="kw"><a href="../reference/plot_maps.html">plot_maps</a></span>(<span class="dt">map_df =</span> map_list[[<span class="dv">1</span>]], <span class="dt">map1_chrom_breaks =</span> map_list[[<span class="dv">2</span>]], <span class="dt">map2_chrom_breaks =</span> map_list[[<span class="dv">3</span>]])</code></pre></div>
<p><img src="syntr_tutorial_files/figure-html/plot_map-1.png" width="576"></p>
<p>The chromosomes are annotated with their species-specific names, an the chromosome breaks are indicated by vertical and horizontal lines.</p>
<p>Notice that the maps look rather scattered. This is because the chromosomes are ordered arbitrarily (there is no inherent order to chromsomes). Top aid readability, we can reorder the order of chromosomes of species 2 to match the order in species 1 (plotted along the x-axis). We also rotate any “wholly-inverted” chromosomes so that they line up (at the chromosome scale) along the 1:1 line.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># reorder and flip some chromosomes</span>
map2_chr_order &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Pet8"</span>, <span class="st">"Pet9"</span>, <span class="st">"Pet12_16"</span>, <span class="st">"Pet15"</span>, <span class="st">"Pet16_17"</span>, <span class="st">"Pet17"</span>)
flip_chrs &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Pet9"</span>)
map_list &lt;-<span class="st"> </span><span class="kw"><a href="../reference/make_one_map.html">make_one_map</a></span>(ann_pet_map, <span class="dt">map1_max_chr_lengths =</span> ann_chr_lengths, <span class="dt">map2_chr_order =</span> map2_chr_order, <span class="dt">flip_chrs =</span> flip_chrs)
<span class="kw"><a href="../reference/plot_maps.html">plot_maps</a></span>(<span class="dt">map_df =</span> map_list[[<span class="dv">1</span>]], <span class="dt">map1_chrom_breaks =</span> map_list[[<span class="dv">2</span>]], <span class="dt">map2_chrom_breaks =</span> map_list[[<span class="dv">3</span>]])</code></pre></div>
<p><img src="syntr_tutorial_files/figure-html/reorder_map-1.png" width="576"></p>
<p>This results in a more sane-looking dot plot. However, there are clearly some translocations (e.g. Pet12_16 x Ann16) that do not conform to the 1:1 line.</p>
</div>
<div id="determine-optimal-turning-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#determine-optimal-turning-parameters" class="anchor"></a>3. Determine optimal turning parameters</h2>
<p>The key tuning parameters of the syntR algorirthm are <code>max_cluster_range</code> and, to a lesser extent, <code>max_nn_dist</code>. <code>max_cluster_range</code> determines the maximum size of clusters of markers that will be collapsed to a singe point for further analysis. It defines the maximum distance that clustered markers can span. Lager values of <code>max_cluster_range</code> will result in more markers being collapsed into a single point.</p>
<p><code>max_nn_dist</code> determines which markers are considered outliers and removed from further analysis. Markers that do not have a neighbour within <code>max_nn_dist</code> are considered outliers. Larger values of <code>max_nn_dist</code> will result in fewer markers being removed.</p>
<p>A simple method of finding the optimal combination of these two parameters is to fit a range of values for both and choose values that maximize coverage (% of the map assigned to synteny blocks) of the two maps and minimizes the number of outliers (“composite” in the plot below is a scaled sum of these metrics).</p>
<p>We provide an internal method for achieving this below. Note it is slightly computationally intensive and may take several minutes to complete.</p>
<p>If you’d like to skip this step, the default values of <code>max_cluster_range = 2</code> and <code>max_nn_dist = 10</code> are likely well suited to most genetics maps.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find best parameter combination</span>
<span class="co"># run find_synteny_blocks with each parameter combination and collect summary statistics</span>

parameter_data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/test_parameters.html">test_parameters</a></span>(map_list, <span class="dt">max_cluster_range_list =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dt">by =</span> <span class="fl">0.5</span>), <span class="dt">max_nn_dist_list =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/seq">seq</a></span>(<span class="dv">10</span>, <span class="dv">50</span>, <span class="dt">by =</span> <span class="dv">10</span>))
<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>

<span class="co">#&gt; Warning: Factor `block` contains implicit NA, consider using</span>
<span class="co">#&gt; `forcats::fct_explicit_na`</span>
<span class="kw"><a href="../reference/plot_summary_stats.html">plot_summary_stats</a></span>(parameter_data[[<span class="dv">1</span>]], <span class="st">"composite"</span>)</code></pre></div>
<p><img src="syntr_tutorial_files/figure-html/find_params-1.png" width="576"></p>
<p>We choose the first (from left to right) composite coverage “peak” in the parameter matrix (see Ostevik et al. for details about this choice).</p>
<p>So, in this case, we select a <code>max_cluster_range</code> of 2 and a <code>max_nn_dsit</code> of 10.</p>
</div>
<div id="identify-synteny-blocks" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-synteny-blocks" class="anchor"></a>4. Identify synteny blocks</h2>
<p>With the two tuning parameters determined, we can run the primary function of syntR: <code><a href="../reference/find_synteny_blocks.html">find_synteny_blocks()</a></code>.</p>
<p>This function runs a set of marker data (contained in the <code>map_list</code> object we created previously) through the syntR error-detection and synteny block identification algorithm described in the Ostevik et al. publication.</p>
<p>It can be run as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># find synteny blocks</span>
synt_blocks &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_synteny_blocks.html">find_synteny_blocks</a></span>(map_list, <span class="dt">max_cluster_range =</span> <span class="dv">2</span>, <span class="dt">max_nn_dist =</span> <span class="dv">10</span>, <span class="dt">plots =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="syntr_tutorial_files/figure-html/find_blocks-1.png" width="576"></p>
<pre><code>#&gt; Warning: Factor `block` contains implicit NA, consider using
#&gt; `forcats::fct_explicit_na`

#&gt; Warning: Factor `block` contains implicit NA, consider using
#&gt; `forcats::fct_explicit_na`</code></pre>
<p><img src="syntr_tutorial_files/figure-html/find_blocks-2.png" width="576"><img src="syntr_tutorial_files/figure-html/find_blocks-3.png" width="576"></p>
<p>When <code>plots=TRUE</code>, the function generates three plots showing the progress of the algorithm. In the first plot, mapping error is smoothed via a pre-clustering step. Next, outliers are flagged and removed. Finally, synteny blocks are identified via the syntR friends-of-friends clustering algorithm.</p>
<p><code><a href="../reference/find_synteny_blocks.html">find_synteny_blocks()</a></code> outputs a list object with containing five data frames:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># print the contents of synt_blocks</span>

<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/lapply">lapply</a></span>(synt_blocks, as_tibble)
<span class="co">#&gt; $marker_df</span>
<span class="co">#&gt; # A tibble: 1,173 x 12</span>
<span class="co">#&gt;    map1_name map1_chr map1_pos map1_posfull map2_name map2_chr map2_pos</span>
<span class="co">#&gt;    &lt;fct&gt;     &lt;fct&gt;       &lt;dbl&gt;        &lt;dbl&gt; &lt;fct&gt;     &lt;fct&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1 HanXRQCh… Ann12       23.3          223. HanXRQCh… Pet12_16     0   </span>
<span class="co">#&gt;  2 HanXRQCh… Ann12       23.3          223. HanXRQCh… Pet12_16     3.66</span>
<span class="co">#&gt;  3 HanXRQCh… Ann12       51.1          251. HanXRQCh… Pet12_16     3.66</span>
<span class="co">#&gt;  4 HanXRQCh… Ann12       25.0          225. HanXRQCh… Pet12_16     3.66</span>
<span class="co">#&gt;  5 HanXRQCh… Ann12       25.0          225. HanXRQCh… Pet12_16     3.66</span>
<span class="co">#&gt;  6 HanXRQCh… Ann12       22.5          223. HanXRQCh… Pet12_16     4.57</span>
<span class="co">#&gt;  7 HanXRQCh… Ann12        9.06         209. HanXRQCh… Pet12_16     4.57</span>
<span class="co">#&gt;  8 HanXRQCh… Ann12       27.4          228. HanXRQCh… Pet12_16     7.22</span>
<span class="co">#&gt;  9 HanXRQCh… Ann12       32.0          232. HanXRQCh… Pet12_16    12.6 </span>
<span class="co">#&gt; 10 HanXRQCh… Ann12       32.0          232. HanXRQCh… Pet12_16    12.6 </span>
<span class="co">#&gt; # … with 1,163 more rows, and 5 more variables: map2_posfull &lt;dbl&gt;,</span>
<span class="co">#&gt; #   cluster &lt;int&gt;, block &lt;fct&gt;, final_block &lt;int&gt;, orientation &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $synteny_blocks_df</span>
<span class="co">#&gt; # A tibble: 41 x 8</span>
<span class="co">#&gt;    block map1_chr map1_start map1_end map2_chr map2_start map2_end</span>
<span class="co">#&gt;    &lt;fct&gt; &lt;fct&gt;         &lt;dbl&gt;    &lt;dbl&gt; &lt;fct&gt;         &lt;dbl&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt;  1 5     Ann08         0.308     3.10 Pet8            0       5.87</span>
<span class="co">#&gt;  2 6     Ann08         7.97     11.6  Pet8           12.1    28.4 </span>
<span class="co">#&gt;  3 7     Ann08        15.4      23.0  Pet8           28.4    38.2 </span>
<span class="co">#&gt;  4 8     Ann08        24.9      28.6  Pet8           38.2    43.6 </span>
<span class="co">#&gt;  5 9     Ann08        32.6      34.8  Pet8           48.0    50.7 </span>
<span class="co">#&gt;  6 10    Ann08        29.1      32.6  Pet8           51.6    59.1 </span>
<span class="co">#&gt;  7 11    Ann08        37.5      42.2  Pet8           59.1    72.5 </span>
<span class="co">#&gt;  8 12    Ann08        42.7      47.8  Pet8           75.2    82.6 </span>
<span class="co">#&gt;  9 14    Ann08        51.1      63.1  Pet8           82.6   104.  </span>
<span class="co">#&gt; 10 29    Ann09         1.96     28.6  Pet9            0      32.4 </span>
<span class="co">#&gt; # … with 31 more rows, and 1 more variable: orientation &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $map1_breaks</span>
<span class="co">#&gt; # A tibble: 35 x 2</span>
<span class="co">#&gt;     start    end</span>
<span class="co">#&gt;     &lt;dbl&gt;  &lt;dbl&gt;</span>
<span class="co">#&gt;  1   3.10   7.97</span>
<span class="co">#&gt;  2  11.6   15.4 </span>
<span class="co">#&gt;  3  23.0   24.9 </span>
<span class="co">#&gt;  4  28.6   29.1 </span>
<span class="co">#&gt;  5  32.6   32.6 </span>
<span class="co">#&gt;  6  34.8   37.5 </span>
<span class="co">#&gt;  7  42.2   42.7 </span>
<span class="co">#&gt;  8  47.8   51.1 </span>
<span class="co">#&gt;  9 112.   117.  </span>
<span class="co">#&gt; 10 128.   137.  </span>
<span class="co">#&gt; # … with 25 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $map2_breaks</span>
<span class="co">#&gt; # A tibble: 35 x 2</span>
<span class="co">#&gt;     start   end</span>
<span class="co">#&gt;     &lt;dbl&gt; &lt;dbl&gt;</span>
<span class="co">#&gt;  1   5.87  12.1</span>
<span class="co">#&gt;  2  28.4   28.4</span>
<span class="co">#&gt;  3  38.2   38.2</span>
<span class="co">#&gt;  4  43.6   48.0</span>
<span class="co">#&gt;  5  50.7   51.6</span>
<span class="co">#&gt;  6  59.1   59.1</span>
<span class="co">#&gt;  7  72.5   75.2</span>
<span class="co">#&gt;  8  82.6   82.6</span>
<span class="co">#&gt;  9 176.   179. </span>
<span class="co">#&gt; 10 194.   194. </span>
<span class="co">#&gt; # … with 25 more rows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $summary_stats</span>
<span class="co">#&gt; # A tibble: 1 x 4</span>
<span class="co">#&gt;   num_blocks map1_coverage map2_coverage n_outliers</span>
<span class="co">#&gt;        &lt;int&gt;         &lt;dbl&gt;         &lt;dbl&gt;      &lt;int&gt;</span>
<span class="co">#&gt; 1         41          383.          567.        205</span></code></pre></div>
<p>These elements are:</p>
<ol style="list-style-type: decimal">
<li><p><code>$marker_df</code> A data frame including the original list of markers along with their synteny block assignments.</p></li>
<li><p><code>$synteny_blocks_df</code> A data frame summarizing the span of each synteny block in each map (in map units) along with their inferred orientation (colinear, inverted, etc.) for synteny blocks with sufficient evidence of directionality.</p></li>
<li><p><code>$map1_breaks</code> A data frame listing the breakpoints between synteny blocks in map1.</p></li>
<li><p><code>$map2_breaks</code> A data frame listing the breakpoints between synteny blocks in map2.</p></li>
<li><p><code>$summary_stats</code> A data frame summarizing the results of the algorithm (coverage etc.)</p></li>
</ol>
<p>We can plot the blocks with their orientation as below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># plot synteny block orientations</span>
synt_blocks[[<span class="dv">1</span>]] <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="../reference/plot_maps.html">plot_maps</a></span>(map_list[[<span class="dv">2</span>]], map_list[[<span class="dv">3</span>]], <span class="dt">col =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"blue"</span>, <span class="st">"grey"</span>, <span class="st">"red"</span>, <span class="st">"black"</span>)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/factor">as.factor</a></span>(.<span class="op">$</span>orientation))], 
              <span class="dt">main =</span> <span class="st">"Synteny block orientation"</span>,  <span class="dt">cex_val =</span> <span class="fl">0.75</span>)</code></pre></div>
<p><img src="syntr_tutorial_files/figure-html/find_blocks3-1.png" width="576"></p>
<p>In this case, blue = inverted, red = colinear and grey/black = unclassified.</p>
</div>
<div id="questions" class="section level2">
<h2 class="hasAnchor">
<a href="#questions" class="anchor"></a>Questions?</h2>
<p>If you have any questions about syntR, please email one of the authors or post an issue on www.github.com/ksamuk/syntR.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#what-is-syntr">What is syntR?</a></li>
      <li><a href="#requirements">Requirements</a></li>
      <li><a href="#input-file">Input File</a></li>
      <li><a href="#example-workflow">Example Workflow</a></li>
      <li><a href="#load-and-format-the-map-data">1. Load and format the map data</a></li>
      <li><a href="#reordering-the-map">2. Reordering the map</a></li>
      <li><a href="#determine-optimal-turning-parameters">3. Determine optimal turning parameters</a></li>
      <li><a href="#identify-synteny-blocks">4. Identify synteny blocks</a></li>
      <li><a href="#questions">Questions?</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Katherine Ostevik, Kieran Samuk.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
